tolstoy/src/syncer.rs

```diff
@@ -143,8 +147,27 @@ impl<'c> TxReceiver for UploadingTxReceiver<'c> {
 
         // TODO separate bits of network work should be combined into single 'future'
 
+        let mut datoms: Vec<TxPart> = datoms.collect();
+
+        // Figure our the "high water-mark" for the user partition.
+        let mut largest_e = USER0;
+        for datom in &datoms {
+            if datom.e > largest_e {
+                largest_e = datom.e;
+            }
+        }
+
+        // Annotate first datom in the series with the user partition information.
+        // TODO this is obviously wrong - we want to read partition info without
```

Comment:
It feels like this is at the wrong place in the stack.  You want to change the flow from `Vec<Tx>` to `(PartitionMap, Vec<Tx>)`, but what you've done is change it to `Vec<(Option<PartitionMap>, Tx)>` and an assumption about option.

It feels to me like we need some notion of a *chunk* that's more general than what we have now; some chunks are transaction data, yes; some are transactions themselves; but some are also partition maps.

I think you're probably aware of this, and tomorrow we can talk through the difficulty of changing the underlying model more broadly.

Reply:
I'm viewing this particular change as purely temporary - I really don't like having partition information one step removed from the transaction - and following the path of least resistance for now. The goal is to move this out of chunks entirely, and lift it up into transactions. They should be annotated with partitionmaps, and the only reason that's not the way it is now is lack of server support.

Currently, I don't think there's any _other_ need to generalize a chunk, and doing so seems like a distraction.

My higher level view of this - let's settle on the overall flow first - how data generally moves around, what do we need to move around, etc - then re-shape the server and clients to match. In the meantime, iterate quickly and maintain some kind of momentum.

"Nothing is as permanent as the temporary", but I hope that won't apply :dancing_men: